### PR TITLE
Update soil layer data erosion attributes from percents to fractions

### LINKED ISSUE #467: Rename and refactor Erosion attributes in `LayerData`
# Problem
The set of attributes (`percent_organic_carbon_content`, `percent_clay_content`, `percent_sand_content`, `percent_silt_content`, and `percent_rock_content`) used by the erosion submodule in the soil module and located in `LayerData` follow the patterns:
1. The name takes the form of `percent_<name>_content`
2. The value represents a percent, and is in the range [0, 100] (1% = 1)
This is an issue because it is not consistent with other modules, and can lead to confusion.

# Solution
Refactor this set of attributes so that they follow the patterns:
1. [ ] The name takes the form of `<name>_proportion_content` (or whatever pattern; as long as it is clear, simple, and consistently used the naming pattern doesn't really matter)
2. [ ] The value should be changed to be in the range [0.0, 1.0] (So that 1% = 0.01)

After these patterns have been implemented, the following tasks need to be accomplished:
- [ ] All docstrings of these attributes need to be refactored indicate clearly what they mean
- [ ] All methods that use these attributes are refactored to adjust for their change in value (and also have their docstrings updated to indicate the scale of the value they take and how they adjust it)

### PR DESCRIPTION
**Pull request by ew3361zh**
Renames and refactors Erosion attributes in LayerData as well as everywhere they're referenced.

### Context

Issue(s) closed by this pull request: closes #467

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.

### What

The set of attributes used by the erosion submodule in the soil module and located in LayerData have been updated from percents to fractions:
`percent_organic_carbon_content` -> `organic_carbon_fraction`
`percent_clay_content` -> `clay_fraction`
`percent_sand_content` -> `sand_fraction`
`percent_silt_content` -> `silt_fraction`
`percent_rock_content` -> `rock_fraction`

The previous format of the attribute was to be as a percent (e.g. 55.2%) and the new fraction version would be (0.552).

All references (in the model, docstrings, in testing) have been updated to reflect these changes as well.

### Why

This is done to be consistent with other parts of the C&S Module as well as other parts of the RuFaS model.

### How

Using search and update for all the references.
Adjusting the equations, references, and docstrings accordingly.
Updating test cases to match new format.

### Test plan

Ran successfully with all metadata scenarios. Testing updated and passing as before.

### CODE REVIEW

File: RUFAS/routines/field/soil/nitrogen_cycling/denitrification.py
Comment (matthew7838) [inline RUFAS/routines/field/soil/nitrogen_cycling/denitrification.py]:
```suggestion
            The fraction of this soil layer that is made up of organic carbon, in the range [0, 1.0] (unitless).
```

File: RUFAS/output_manager.py
Comment (matthew7838) [inline RUFAS/output_manager.py]:
```suggestion
        units : str | Dict[str, str] | None
            The units associated with the data.
        subkey : str, optional, default None
```

Comment (matthew7838) [reply RUFAS/output_manager.py]: Is the unit a string or a dictionary?

File: RUFAS/routines/field/soil/nitrogen_cycling/denitrification.py
Comment (matthew7838) [inline RUFAS/routines/field/soil/nitrogen_cycling/denitrification.py]: This should be something that I missed while reformatting it. They're all missing periods.

File: RUFAS/output_manager.py
Comment (matthew7838) [inline RUFAS/output_manager.py]:
```suggestion
            csv_column_lists: Dict[str, List[Any]] = {
            subkey: [] 
            for item in data_list 
            for subkey in item.keys()
}
```
More readable if this is written in multiple lines.

**Review by matthew7838 — APPROVED** [review]
LGTM, there are some undocumented changes in the templates. If those were meant to be changed then it's all good. Great work Niko.

File: RUFAS/output_manager.py
Comment (ew3361zh) [reply RUFAS/output_manager.py]: Ugh, this unfortunately isn't my code. It looks like I may have accidentally rebased incorrectly and pulled in changes and commits from another PR.

File: input/data/soil/default_soil.json
Comment (Isgotkowitz) [inline input/data/soil/default_soil.json]:
```suggestion
            "organic_carbon_fraction": 0.012,
```

File: input/data/soil/default_soil_2.json
Comment (Isgotkowitz) [inline input/data/soil/default_soil_2.json]:
```suggestion
            "organic_carbon_fraction": 0.00348,
```

File: input/data/soil/default_soil_2.json
Comment (Isgotkowitz) [inline input/data/soil/default_soil_2.json]:
```suggestion
            "organic_carbon_fraction": 0.00145,
```

File: input/data/soil/default_soil_2.json
Comment (Isgotkowitz) [inline input/data/soil/default_soil_2.json]:
```suggestion
            "organic_carbon_fraction": 0.00145,
```

File: input/metadata/properties/default.json
Comment (Isgotkowitz) [inline input/metadata/properties/default.json]:
```suggestion
          "default": 0.012
```

File: tests/soil_crop_tests/soil_tests/test_soil_erosion.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/soil_tests/test_soil_erosion.py]: Maybe rename this to rock_fraction

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.20,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.20,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.20,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.1304,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.2233,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.2727,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                    "clay_fraction": 0.2195,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                clay_fraction=0.2727,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                "clay_fraction": 0.2727,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                clay_fraction=0.2195,
```

File: tests/soil_crop_tests/manager_tests/test_field_manager.py
Comment (Isgotkowitz) [inline tests/soil_crop_tests/manager_tests/test_field_manager.py]:
```suggestion
                "clay_fraction": 0.2195,
```

**Review by Isgotkowitz — APPROVED** [review]
Noticed some spots in inputs and tests where the percentage had not been correctly converted to a fraction, should be a simple fix. Am commenting for right now, I want to do some more extensive comparison between this branch and main to make sure there aren't any unreasonable changes in erosion amounts or carbon levels. I will come back and approve once I do this.

File: RUFAS/routines/field/soil/soil_erosion.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
        not_sand_fraction = 1 - sand_fraction
```

File: RUFAS/routines/field/soil/soil_erosion.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]:
```suggestion
            raise ValueError("Cannot have fractions of silt and clay in the soil both be 0")
```

File: RUFAS/routines/field/soil/soil_erosion.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_erosion.py]: This method still needs to be updated to account for the switch to fractions.

**Review by Isgotkowitz — COMMENTED** [review]
Found a few places that need to be updated to account of the use of fractions, and not percents. Also, am still seeing minor differences in the default between this branch and main, looking with this filter:
```
{
    "multiple": [
        {
            "name": "Carbon",
            "filters": [
                "FieldDataReporter.send_daily_variables.profile_carbon_total.field='field'"
            ],
            "vertical_aggregation": "average"
        },
        {
            "name": "Erosion",
            "filters": [
                "FieldDataReporter.send_daily_variables.eroded_sediment.field='field'"
            ],
            "vertical_aggregation": "average"
        }
    ]
}
```
I think it would be good to get absolutely no difference between all scenarios before merging, or at least make sure the differences between all scenarios are really small (like >1%)

**Comment by krb78** [comment]
After scouring the code and looking for any reasons there might be these slight differences when changing from % to fraction, I cannot identify any reasons for the differences in soil carbon. Given the differences are so minor (<1% at the end of the entire simulation), I think we are ok to accept this as a rounding difference in the model. While we do work for the highest level of accuracy possible, even the best carbon soil researchers don't have confidence in their measured data as precisely as 20kg/ha (the margins of error are bigger than that).

**Pull request merged**
